airflow/providers/amazon/aws/hooks/emr.py

```diff
@@ -68,22 +68,24 @@ def get_cluster_id_by_name(self, emr_cluster_name: str, cluster_states: list[str
         :param emr_cluster_name: Name of a cluster to find
         :param cluster_states: State(s) of cluster to find
         :return: id of the EMR cluster
-        """
+        """        
         response = self.get_conn().list_clusters(ClusterStates=cluster_states)
+        while true:
+            matching_clusters = list(
+                filter(lambda cluster: cluster["Name"] == emr_cluster_name, response["Clusters"])
+            )
 
-        matching_clusters = list(
-            filter(lambda cluster: cluster["Name"] == emr_cluster_name, response["Clusters"])
-        )
-
-        if len(matching_clusters) == 1:
-            cluster_id = matching_clusters[0]["Id"]
-            self.log.info("Found cluster name = %s id = %s", emr_cluster_name, cluster_id)
-            return cluster_id
-        elif len(matching_clusters) > 1:
-            raise AirflowException(f"More than one cluster found for name {emr_cluster_name}")
-        else:
-            self.log.info("No cluster found for name %s", emr_cluster_name)
-            return None
+            if len(matching_clusters) == 1:
```

Comment:
This logic isn't quite right, what if two clusters have the same name but are on different pages?

Reply:
Addressed this in latest commit. I missed that cluster_name was non-unique.